### PR TITLE
Change Objective-C API to chained call

### PR DESCRIPTION
**Pull request by protosse**
I have always used PinLayout with Swift before, but lately i have to begin a project with Objective-C.
Too many brackets are really a nightmare.

### CODE REVIEW

File: Example/PinLayoutSample/UI/Examples/IntroObjectiveC/IntroObjectiveCView.m [hunk 1]
```diff
@@ -68,10 +68,30 @@ - (void) layoutSubviews {
...
     
-    [[[[[[logo.pinObjc topWithInsets:safeArea] leftWithInsets:safeArea] width:100] aspectRatio] margin:margin] layout];
-    [[[[segmented.pinObjc rightOf:logo aligned:VerticalAlignTop] rightWithInsets:safeArea] marginHorizontal:margin] layout];
-    [[[[[[textLabel.pinObjc belowOf:segmented aligned:HorizontalAlignLeft] widthOf:segmented] pinEdges] marginTop:margin] sizeToFit:FitWidth] layout];
-    [[[[[separatorView.pinObjc belowOfViews:@[logo, textLabel] aligned:HorizontalAlignLeft] rightTo:segmented.edge.right] height:1] marginTop:margin] layout];
+//    [[[[[[logo.pinObjc topWithInsets:safeArea] leftWithInsets:safeArea] width:100] aspectRatio] margin:margin] layout];
+//    [[[[segmented.pinObjc rightOf:logo aligned:VerticalAlignTop] rightWithInsets:safeArea] marginHorizontal:margin] layout];
+//    [[[[[[textLabel.pinObjc belowOf:segmented aligned:HorizontalAlignLeft] widthOf:segmented] pinEdges] marginTop:margin] sizeToFit:FitWidth] layout];
+//    [[[[[separatorView.pinObjc belowOfViews:@[logo, textLabel] aligned:HorizontalAlignLeft] rightTo:segmented.edge.right] height:1] marginTop:margin] layout];
+    logo.pinObjc
```
Comment (lucdion) [inline Example/PinLayoutSample/UI/Examples/IntroObjectiveC/IntroObjectiveCView.m:75]: Could you put this on a single line, so that it matches Swift style?

**Review by lucdion — COMMENTED** [review]
Wow, really nice, I always wanted to support that, but never took the time to do it. Really nice job 👌👌👌

One tine comment + could you update the documentation (docs/objective_c.md) 
Thanks

File: Example/PinLayoutSample/UI/Examples/IntroObjectiveC/IntroObjectiveCView.m (hunk 1 above)
Comment (protosse) [reply Example/PinLayoutSample/UI/Examples/IntroObjectiveC/IntroObjectiveCView.m]: OK👌

**Comment by protosse** [comment]
image
sorry i don't known why it build failed on macOS, it seems to be running well.

**Comment by ingod9true** [comment]
合并好了吗，大佬

**Comment by lucdion** [comment]
@protosse, you need to fix the compilation of the objective-c unit tests (`ObjectiveCSpec.m`)

File: docs/objective_c.md [hunk 2]
```diff
@@ -47,6 +50,6 @@ When using the Objective-c interface, the `layout` method must be called explici
  view.pin.width(100)
 
  // Objective-c
- [[view.pinObjc width:100] layout];
+ view.pinObjc.width().layout();
```
Comment (lucdion) [inline docs/objective_c.md:53]:
```suggestion
 view.pinObjc.width(100).layout();
```

**Review by lucdion — CHANGES_REQUESTED** [review]
You just need to fix unit tests + this little doc change

**Comment by protosse** [comment]
@lucdion hello, i have fixed all the compilation of the objective-c unit tests, except the code in the carthage directory, it will use the PinLayout from github, so can i comment them?
image

**Comment by lucdion** [comment]
@protosse yes its fine for carthage. You should alse update your branch with the latest master, I have update to project to Xcode 12.5 and fix the build scripts.

**Review by lucdion — APPROVED** [review]
Excellent job 👌

**Comment by lucdion** [comment]
I have published this new release https://github.com/layoutBox/PinLayout/releases/tag/1.10.0 that includes this PR.
Thanks again!

**Pull request merged**